src/igvfd/schemas/analysis_step.json

```diff
@@ -0,0 +1,139 @@
+        }
+    ],
+    "dependentSchemas": {},
+    "properties": {
+        "schema_version": {
+            "default": "1"
+        },
+        "analysis_step_types": {
+            "title": "Analysis Step Types",
+            "description": "The classification of the software.",
+            "type": "array",
+            "uniqueItems": true,
+            "items": {
+                "title": "Type",
+                "type": "string",
+                "enum": [
+                    "alignment",
+                    "file format conversion",
+                    "signal generation"
+                ]
+            }
+        },
+        "step_label": {
+            "title": "Step Label",
+            "description": "Unique lowercased label of the analysis step that includes the relevant assays, the software used, and the purpose of the step, e.g. rampage-grit-peak-calling-step",
+            "comment": "Unique for each analysis_step in a given workflow.",
+            "type": "string",
+            "pattern": "^[a-z0-9-]+-step$"
+        },
+        "title": {
+            "title": "Title",
+            "description": "The preferred viewable name of the analysis step, likely the same as the step label.",
+            "type": "string",
+            "pattern": "^[a-zA-Z\\d_().,-]+(?:\\s[a-zA-Z\\d_().,-]+)*[step|Step]$"
+        },
+        "workflow": {
+            "title": "Workflow",
+            "description": "The computational workflow in which this analysis step belongs.",
+            "type": "string",
+            "linkTo": "Workflow"
+        },
+        "parents": {
+            "title": "Parents",
+            "description": "The precursor steps.",
+            "uniqueItems": true,
+            "type": "array",
+            "items": {
+                "description": "A precursor step.",
+                "comment": "See analysis_step.json for available identifiers.",
+                "type": "string",
+                "linkTo": "AnalysisStep"
+            }
+        },
+        "input_content_types": {
+            "title": "Input Content Types",
+            "description": "The content types used as input for the analysis step.",
+            "type": "array",
+            "uniqueItems": true,
+            "items": {
+                "title": "Input Content Type",
+                "description": "A content type used as input for the analysis step.",
+                "$merge": "#/$defs/content_types"
+            }
+        },
+        "output_content_types": {
+            "title": "Output Content Types",
+            "description": "The content types produced as output by the analysis step.",
+            "type": "array",
+            "uniqueItems": true,
+            "items": {
+                "title": "Output Content Type",
+                "description": "A content type produced as output by the analysis step.",
+                "$merge": "#/$defs/content_types"
+            }
+        }
+    },
+    "$defs": {
```

Comment:
do we want to do something more extensive - like collecting content types from all the file subtypes?

Reply:
I wasn't sure how to do that. Is there a way I could reference content types in other schemas?